ja/core-libraries/logging.rst

```diff
@@ -1,23 +1,387 @@
-Logging
-#######
+ロギング
+########
 
-.. note::
-    The documentation is not currently supported in Japanese language for this
-    page.
+CakePHP コアクラスの Configure 設定は、内部で何が起きているかを知るための有益な手段です。
+そこで、何が起きているかをディスクにログデータとして保存する必要が出てくるでしょう。
+SOAPやAJAX、REST APIのような技術を一緒に使うとデバッグはむしろ難しくなるでしょう。
 
-    Please feel free to send us a pull request on
-    `Github <https://github.com/cakephp/docs>`_ or use the **Improve This Doc**
-    button to directly propose your changes.
+ロギングは、時系列でアプリケーションで何が起きているかを知るための手段です。
+何の検索ワードが使われましたか？何のエラーがユーザーに表示されましたか？
+どのくらいの頻度で特定のクエリが実行されましたか？
 
-    You can refer to the English version in the select top menu to have
-    information about this page's topic.
+CakePHP でデータのロギングは簡単です。
+log() 関数は ``LogTrait`` により提供されており、どれも多くの CakePHP クラス向けの一般的な祖先です。
+もし、環境がCakePHP のクラス( Controller や Component, View 等)であれば、あなたはデータを記録することができます。
+直接 ``Log::write()`` を同様に使うことも出来ます。
+:ref:`writing-to-logs` を参照してください。
 
 .. _log-configuration:
 
-Logging Configuration
-=====================
+ロギング設定
+============
+
+アプリケーションの起動処理の間に、``Log`` の設定は完了しているはずで
+**config/app.php** は、これのためです。
+多かれ少なかれアプリケーションが望むロガーを定義できます。
+ロガーは、 :php:class:`Cake\\Core\\Log` を使い設定する必要があります。
+一例として ::
+
+    use Cake\Log\Log;
+
+    // 短いクラス名場合
+    Log::config('debug', [
+        'className' => 'File',
+        'path' => LOGS,
+        'levels' => ['notice', 'info', 'debug'],
+        'file' => 'debug',
+    ]);
+
+    // namespaceからまるまるの場合
+    Log::config('error', [
+        'className' => 'Cake\Log\Engine\FileLog',
+        'path' => LOGS,
+        'levels' => ['warning', 'error', 'critical', 'alert', 'emergency'],
+        'file' => 'error',
+    ]);
+
+上記では二つのロガーを作っています。  一つは  ``debug`` で、その他は ``error`` です。
+異なるレベルのメッセージをハンドルするために、それぞれを設定しました。
+あれらもまた、あれらのログメッセージを別々のファイルとして保持します、
+つまり、エラーの重要性からdebug/notice/infoとして簡単に分けられます。
+色々なレベルに応じた情報と意味の一覧は :ref:`logging-levels` を参照してください。
+
+一度設定を作成した後、それを変更することはできません。
+代わりに、設定を削除し :php:meth:`Cake\\Log\\Log::drop()` と :php:meth:`Cake\\Log\\Log::config()` を使い再作成するべきです。
+
+それらもまた、クロージャーを与えることによりロガーを作成することが可能です。
+これはロガーオブジェクトを構築し、あなたが思い通りにコントロールしたいとき役立ちます。クロージャーは、ロガーインスタンスを構築し返します。例えば::
+
+    Log::config('special', function () {
+        return new \Cake\Log\Engine\FileLog(['path' => LOGS, 'file' => 'log']);
+    });
+
+設定オプションは :term:`DSN` の文字列で渡すことも可能背です。
+これは開発環境や :term:`PaaS` で作業しているときに役立ちます。::
+
+    Log::config('error', [
+        'url' => 'file:///?levels[]=warning&levels[]=error&file=error',
+    ]);
+
+.. note::
+
+    ロガーは ``Psr\Log\LoggerInterface`` を、実装する必要があります。
+
+ログアダプターの作成
+--------------------
+
+ログアダプターはアプリケーションの一部やプラグインの一部になりえます。
+例えばデータベースロガーが ``DatabaseLog`` という名前である場合、
+アプリケーションの一部として **src/Log/Engine/DatabaseLog.php** にセットされ、
+プラグインの一部として **plugins/LoggingPack/src/Log/Engine/DatabaseLog.php** にセットされます。
```

Comment:
> If for example you had a database logger called ``DatabaseLog``.
As part of your application it would be placed in **src/Log/Engine/DatabaseLog.php**.
As part of a plugin it would be placed in **plugins/LoggingPack/src/Log/Engine/DatabaseLog.php**. 

例えば ``DatabaseLog`` という名前のデータベースロガーがあったとします。
アプリケーションの一部としては **src/Log/Engine/DatabaseLog.php** に置かれます。
プラグインの一部としては **plugins/LoggingPack/src/Log/Engine/DatabaseLog.php** に置かれます。